packages/host/app/services/billing-service.ts

```diff
@@ -0,0 +1,11 @@
+import Service from '@ember/service';
+
+import ENV from '@cardstack/host/config/environment';
+const { stripePaymentLink } = ENV;
+
+export default class BillingService extends Service {
+  getStripePaymentLink(matrixUserId: string): string {
+    const clientReferenceId = Buffer.from(matrixUserId).toString('base64');
```

Comment:
There are cases where base64 will add `==` padding, for example 

`Buffer.from("@james:ok.com").toString('base64') => 'QGphbWVzOm9rLmNvbQ=='`
﻿
Given Stripe has limitations on the format of this value (as I recall it only accepts alphanumeric), will this work? 

Also if you encode this value here, then you must decode it in `handleCheckoutSessionCompleted`. I already mentioned this in my previous comments.

Did you test this flow in development, meaning clicking on the button that takes you to Stripe where you subscribe to a free plan and go back to the app and finish registration? It's important we test it this way.

Reply:
@jurgenwerk thanks for reminding me again, I updated the encoding logic to be URL-safe

Also, I have updated the screenshot video testing the complete flow